proto/coroutine/v1/type.proto

```diff
@@ -0,0 +1,115 @@
+syntax = "proto3";
+
+package coroutine.v1;
+
+// Type is a data type.
+//
+// Types may reference other types internally (including themselves).
+// To encode the graph of types in a program, a Type is expected to be
+// stored alongside all other Types in an array. The index of each Type
+// in the array becomes its unique identifier, and internal references
+// to other types are simply an int32 index.
+message Type {
+  // Name is the name of the type within the package it was defined,
+  string name = 1;
+
+  // Package is the name of the package that defines the type.
+  string package = 2;
+
+  // Kind is the underlying type.
+  Kind kind = 3;
+
+  // Elem is the type of an array, slice, pointer, chan, or map's element.
+  int32 elem = 4;
+
+  // Key is the key type for map types.
+  int32 key = 5;
+
+  // Fields is the set of fields defined in a struct type.
+  repeated Field fields = 6;
+
+  // Params are the types of params for a function type.
+  repeated int32 params = 7;
+
+  // Results are the types of results for a function type.
+  repeated int32 results = 8;
+
+  // Length is the length of an array type.
+  int64 length = 9;
+
+  // Variadic is true for function types with a variadic argument.
+  bool variadic = 10;
```

Comment:
How about grouping the booleans? That'll reduce the padding needed to align struct fields in memory a bit. 

Or maybe we can find a tool that will do this for us?

Reply:
See 5dd2e8c410cd9356c7feb7325232c678ecbd2147